tests/integration/devfile/cmd_devfile_push_test.go

```diff
@@ -409,6 +409,27 @@ var _ = Describe("odo devfile push command tests", func() {
 			}
 			Expect(volumesMatched).To(Equal(true))
 		})
+
+		It("Ensure that push -f correctly removes local deleted files from the remote target sync folder", func() {
+
+			// 1) Push a generic Java project
+			helper.CmdShouldPass("odo", "create", "java-springboot", "--project", namespace, cmpName)
+			helper.CopyExample(filepath.Join("source", "devfiles", "springboot", "project"), context)
+
+			output := helper.CmdShouldPass("odo", "push", "--namespace", namespace)
+			Expect(output).To(ContainSubstring("Changes successfully pushed to component"))
+
+			// 2) Rename the pom.xml, which should cause the build to fail if sync is working as expected
+			err := os.Rename(filepath.Join(context, "pom.xml"), filepath.Join(context, "pom.xml.renamed"))
+			Expect(err).NotTo(HaveOccurred())
+
+			// 3) Ensure that the build fails due to missing 'pom.xml', which ensures that the sync operation
+			// correctly renamed pom.xml to pom.xml.renamed.
+			session := helper.CmdRunner("odo", "push", "-v", "5", "-f", "--namespace", namespace)
+			helper.WaitForOutputToContain("Non-readable POM", 180, 10, session)
```

Comment:
The wait is done on ```session.Out.Contents()``` however i was expecting the wait should be done on ```session.Err.Contents()``` instead as it fails. Is it by design ```session := helper.CmdRunner("odo", "push", "-v", "5", "-f", "--namespace", namespace)``` always returns to stdout ?

Reply:
Hi @amitkrout , yes this is by design: the `Non-readable POM` message is coming from `mvn` command executed from  [here](https://github.com/openshift/odo/blob/3f3d8b970e9fa3693e319dd27a6adeb84a5ac1bc/pkg/exec/exec.go#L64) which is stdout-only.

Reply:
Cool!!!
Thanks